examples/bail_ensure fails no_std test

```
$ cargo test --no-default-features
   Compiling failure v0.1.1 (file:///home/travis/build/withoutboats/failure)
error[E0432]: unresolved import `failure::Error`
 --> examples/bail_ensure.rs:4:5
  |
4 | use failure::Error;
  |     ^^^^^^^^^^^^^^ no `Error` in the root
error: cannot find macro `bail!` in this scope
 --> examples/bail_ensure.rs:8:5
  |
8 |     bail!("ruh {}", "roh");
  |     ^^^^
error: cannot find macro `ensure!` in this scope
  --> examples/bail_ensure.rs:12:5
   |
12 |     ensure!(true, "true is false");
   |     ^^^^^^
error: cannot find macro `ensure!` in this scope
  --> examples/bail_ensure.rs:13:5
   |
13 |     ensure!(false, "false is false");
   |     ^^^^^^
error: aborting due to 4 previous errors
error: Could not compile `failure`.
warning: build failed, waiting for other jobs to finish...
error: build failed
The command "cargo test --no-default-features" exited with 101.
```

Somehow this example needs to only be tested with the `std` feature. I don't know how to do this at this time.

Discovered in #113.

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,3 +5,6 @@ rust:
     - beta
     - nightly
 cache: cargo
+script:
+- cargo test
+- cargo test --no-default-features

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,3 +20,8 @@ version = "0.3.3"
 default = ["std", "derive"]
 std = ["backtrace"]
 derive = ["failure_derive"]
+
+[[example]]
+name = "bail_ensure"
+path = "./examples/bail_ensure.rs"
+required-features = ["std"]

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -205,6 +205,7 @@ impl Fail {
 #[cfg(feature = "std")]
 impl<E: StdError + Send + Sync + 'static> Fail for E {}
 
+#[cfg(feature = "std")]
 impl Fail for Box<Fail> {
     fn cause(&self) -> Option<&Fail> {
         (**self).cause()

